src/Neo/SmartContract/Manifest/ExtendedType.cs

```diff
@@ -0,0 +1,331 @@
+        /// If used for InteropInterface it contains the type of an individual iterator's value. If used for Map it contains map value type.
+        /// If this field is used, fields MUST NOT be present.
+        /// </summary>
+        public ExtendedType? Value { get; set; }
+
+        /// <summary>
+        /// fields is used for Array type and when used it means that the type is a structure (ordered set of named values of diffent types),
+        /// which has its fields defined directly here (unlike namedtype which is just a reference to namedtypes).
+        /// It's an array with each member being a Parameter. fields MUST NOT be used in method parameter or return value definitions
+        /// (these MUST use namedtype referring to a valid type specified in the namedtypes object).
+        /// </summary>
+        public ContractParameterDefinition[]? Fields { get; set; }
+
+        void IInteroperable.FromStackItem(StackItem stackItem)
+        {
+            FromStackItem((VM.Types.Array)stackItem, 0);
+        }
+
+        internal void FromStackItem(VM.Types.Array array, int startIndex)
+        {
+            Type = (ContractParameterType)(byte)array[startIndex++].GetInteger();
+            if (!Enum.IsDefined(typeof(ContractParameterType), Type)) throw new FormatException();
+            NamedType = array[startIndex++].GetString();
+
+            if (array[startIndex++] is Integer length)
+            {
+                Length = checked((int)length.GetInteger());
+                if (Length < 0) throw new FormatException("Length must be non-negative.");
+                if (Length > ExecutionEngineLimits.Default.MaxItemSize) throw new FormatException($"Length must less than {ExecutionEngineLimits.Default.MaxItemSize}.");
+            }
+            else
+            {
+                Length = null;
+            }
+
+            if (array[startIndex++] is VM.Types.Boolean forbidnull)
+            {
+                ForbidNull = forbidnull.GetBoolean();
+            }
+            else
+            {
+                ForbidNull = null;
+            }
+
+            if (array[startIndex++] is ByteString interf)
+            {
+                if (!Enum.TryParse<Nep25Interface>(interf.GetString(), false, out var inferValue))
+                    throw new FormatException();
+
+                Interface = inferValue;
+            }
+            else
+            {
+                Interface = null;
+            }
+
+            if (array[startIndex++] is ByteString key)
+            {
+                if (!Enum.TryParse<ContractParameterType>(key.GetString(), false, out var keyValue))
+                    throw new FormatException();
+
+                Key = keyValue;
+            }
+            else
+            {
+                Key = null;
+            }
+
+            if (array[startIndex++] is Struct value)
+            {
+                Value = new ExtendedType();
+                Value.FromStackItem(value, 0);
+            }
```

Comment:
It's not specified in NEP-25 directly, but do we need an explicit depth restriction for `Value`? It may contain nested extended types.

From one hand, we have a reference counter that manages maximum possible size of nested types, and it should be sufficient. From another hand, for some nested types like witness conditions we have an explicit depth restriction.

Reply:
We can add a max depth if you consider, otherwise, reference counter will help